### PR TITLE
Posts refactor & stations/observations display

### PR DESCRIPTION
**Pull request by ya-web**

### CODE REVIEW

File: src/Controller/PostsController.php [hunk 1]
```diff
@@ -47,16 +52,15 @@ public function articlePage(
     ) {
         $articleRepository = $this->getDoctrine()->getRepository(Post::class)->setPosts('article');
         $article = $articleRepository->findBySlug($slug);
-        // @TODO: redirect to 404 page
         if (null === $article) {
-            return $this->index($manager);
+            throw new NotFoundHttpException('La page demandée n\'existe pas');
```
Comment (kstefanini) [inline src/Controller/PostsController.php]: Pro tip : use **altgr + G** to print real apostrophe (and avoid slash usage) : `’`

File: templates/components/obs_chips.html.twig [hunk 2]
```diff
@@ -10,6 +10,7 @@
             stade*: string
             date*: string
             author*: int
+    {{ dump(observations) }}
```
Comment (kstefanini) [inline templates/components/obs_chips.html.twig]: dump :)

File: templates/components/calendar-tab.html.twig [hunk 3]
```diff
@@ -26,31 +26,28 @@
     'Feuillaison',
     'Floraison',
     'Fructification',
-    'Scénescence'
+    'Sénescence'
```
Comment (kstefanini) [inline templates/components/calendar-tab.html.twig]: You sould store these labels in class constant.
Twig usage be like :
`{{ set stades_labels = constant('App\\Entity\\Evenement::display_labels) }}`

File: src/Repository/EspeceRepository.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            ;
+            $evenementNom = ucfirst($evenement->getNom());
+            $StadeBbch = $evenement->getStadeBbch();
+
+            if (!in_array($evenementNom, $loopEvenementNom)) {
+                $loopEvenementNom[] = $evenementNom;
+                $loopStadeBbch[] = $StadeBbch;
+                $stades[$evenementNom] = [date_format($evenementEspece->getDateDebut(), 'Y-m-d')];
+            }
+            if (in_array($evenementNom, ['1ere apparition', 'Fructification']) || !in_array($StadeBbch, $loopStadeBbch)) {
```
Comment (kstefanini) [inline src/Repository/EspeceRepository.php]: Prefer class constant usage, eg:
`in_array($evenementNom, Evenement::uniqEvent)`

File: src/Repository/EspeceRepository.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            ->findBy(['espece' => $id], ['espece' => 'asc'])
+        ;
+        $loopEvenementNom = [];
+        $loopStadeBbch = [];
+        $stades = [];
+        foreach ($evenementsEspeces as $evenementEspece) {
+            $evenement = $manager->getRepository(Evenement::class)
+                ->find($evenementEspece->getEvenement())
+            ;
+            $evenementNom = ucfirst($evenement->getNom());
```
Comment (kstefanini) [inline src/Repository/EspeceRepository.php]: Prefer class constant for events name/label

File: templates/components/obs_chips.html.twig (hunk 2 above)
Comment (ya-web) [reply templates/components/obs_chips.html.twig]: This was comment

**Pull request merged**